In [24]:
import pandas as pd
from scipy import stats
import numpy as np

shoppers_data = pd.read_csv('enriched_shoppers_data.csv')

In [25]:
# A/B test for weekend vs weekday shoppers

# Weekend vs Weekday Conversion Rate
weekend = shoppers_data[shoppers_data['Weekend'] == True]
weekday = shoppers_data[shoppers_data['Weekend'] == False]

# Success counts
success_weekend = weekend['Revenue'].sum()
success_weekday = weekday['Revenue'].sum()

# Sample sizes
n_weekend = len(weekend)
n_weekday = len(weekday)

# Proportion test
z_stat, p_value = proportions_ztest(
    count=[success_weekend, success_weekday],
    nobs=[n_weekend, n_weekday]
)

print(f"Z-Stat: {z_stat:.3f}, p-value: {p_value:.4f}")


weekend = shoppers_data[shoppers_data['Weekend'] == True]
weekday = shoppers_data[shoppers_data['Weekend'] == False]

# Success counts
success_weekend = weekend['Revenue'].sum()
success_weekday = weekday['Revenue'].sum()

# Sample sizes
n_weekend = len(weekend)
n_weekday = len(weekday)

# Proportion test
z_stat, p_value = proportions_ztest(
    count=[success_weekend, success_weekday],
    nobs=[n_weekend, n_weekday]
)

print(f"Z-Stat: {z_stat:.3f}, p-value: {p_value:.4f}")


Z-Stat: 3.063, p-value: 0.0022
Z-Stat: 3.063, p-value: 0.0022


In [26]:
# Bounce Rates by Conversion – t-Test
converted = shoppers_data[shoppers_data['Revenue'] == True]['BounceRates']
non_converted = shoppers_data[shoppers_data['Revenue'] == False]['BounceRates']

t_stat, p_val = stats.ttest_ind(converted, non_converted, equal_var=False)
print(f"T-Statistic: {t_stat:.3f}, p-value: {p_val:.4f}")


T-Statistic: -32.687, p-value: 0.0000


In [27]:
#Conversion Rate by Interest Category – ANOVA
groups = [group['Revenue'] for name, group in shoppers_data.groupby('interest_category')]
f_stat, p_val = stats.f_oneway(*groups)
print(f"F-statistic: {f_stat:.3f}, p-value: {p_val:.4f}")


F-statistic: 1.761, p-value: 0.1523


In [28]:
#Engagement Score vs Revenue – t-Test
engaged = shoppers_data[shoppers_data['Revenue'] == True]['engagement_score']
not_engaged = shoppers_data[shoppers_data['Revenue'] == False]['engagement_score']

t_stat, p_val = stats.ttest_ind(engaged, not_engaged, equal_var=False)
print(f"T-Statistic: {t_stat:.3f}, p-value: {p_val:.4f}")


T-Statistic: 14.706, p-value: 0.0000


1) Weekend vs Weekday Conversion – A/B Test (Z-Test)
I compared the conversion rates between users who visited the site on weekends and those who visited during weekdays. The p-value was 0.0022, which is below the standard 0.05 threshold for significance.

Interpretation:
This means that weekend visitors are significantly more likely to make a purchase compared to weekday visitors. Although the difference in proportions isn't huge, it is statistically meaningful.

2) Bounce Rates and Conversion – Two-Sample t-Test
Next, I compared the average bounce rates of users who converted (made a purchase) and those who didn’t. The p-value was 0.0000, indicating an extremely significant result.

Interpretation:
Users who made a purchase had significantly lower bounce rates than those who did not. This confirms that a high bounce rate is a strong negative signal for purchase intent.

3) Conversion by Interest Category – ANOVA
I used a one-way ANOVA test to compare conversion rates across different interest categories (e.g., Product-Focused, Research-Oriented, Balanced Shoppers). The p-value was 0.1523, which is above 0.05.

Interpretation:
Although some trends were visible in the exploratory analysis, the differences between interest categories were not statistically significant. This suggests that user interest types alone may not strongly predict conversion on their own.

4) Engagement Score and Conversion – Two-Sample t-Test
Finally, I tested whether engagement scores differed between users who converted and those who did not. The p-value was again 0.0000, showing a highly significant difference.

Interpretation:
Users with higher engagement scores were much more likely to convert. This supports the idea that the engagement metric I designed is a meaningful predictor of conversion behavior.

5) Summary
Out of the four tests, three produced statistically significant results:

Conversion rates are higher on weekends,

Users who bounce less convert more,

Higher engagement scores are linked to conversions.

The only non-significant result was the difference across interest categories. Overall, the tests validate several behavioral features introduced during feature engineering and support their use in building predictive models in the next phase of the project.
